xrpl/model/transactions/validations_xrpl_objects.go

```diff
@@ -88,11 +74,19 @@ func IsAmount(amount interface{}) bool {
 }
 
 // IsIssuedCurrency checks if the given object is a valid IssuedCurrency object.
-func IsIssuedCurrency(input map[string]interface{}) bool {
-	return len(maputils.GetKeys(input)) == ISSUED_CURRENCY_SIZE &&
-		typecheck.IsString(input["value"]) &&
-		typecheck.IsString(input["issuer"]) &&
-		typecheck.IsString(input["currency"])
+func IsIssuedCurrency(input interface{}) bool {
+	// Check if the input is a map.
+	if !typecheck.IsMap(input) {
```

Comment:
We should remove this validation since we are also casting the type on the line 84